pkg/leakybucket/overflows.go

```diff
@@ -1,135 +1,107 @@
 package leakybucket
 
 import (
-	"encoding/json"
 	"fmt"
 	"net"
 	"strconv"
 
 	"github.com/crowdsecurity/crowdsec/pkg/types"
 )
 
-func FormatOverflow(l *Leaky, queue *Queue) types.SignalOccurence {
-	var am string
+// for now return the struct directly in order to compare between returned struct
+func NewSource(evt types.Event, l *Leaky) types.Source {
+	src := types.Source{}
+	if _, ok := evt.Meta["source_ip"]; ok {
+		source_ip := evt.Meta["source_ip"]
+		src.Ip = net.ParseIP(source_ip)
+		if v, ok := evt.Enriched["ASNNumber"]; ok {
+			src.AutonomousSystemNumber = v
+		}
+		if v, ok := evt.Enriched["IsoCode"]; ok {
+			src.Country = v
+		}
+		if v, ok := evt.Enriched["ASNOrg"]; ok {
+			src.AutonomousSystemOrganization = v
+		}
+		if v, ok := evt.Enriched["Latitude"]; ok {
+			src.Latitude, _ = strconv.ParseFloat(v, 32)
+		}
+		if v, ok := evt.Enriched["Longitude"]; ok {
+			src.Longitude, _ = strconv.ParseFloat(v, 32)
+		}
+		if v, ok := evt.Meta["SourceRange"]; ok {
+			_, ipNet, err := net.ParseCIDR(v)
+			if err != nil {
+				l.logger.Errorf("Declared range %s of %s can't be parsed", v, src.Ip.String())
+			} else if ipNet != nil {
+				src.Range = *ipNet
+				l.logger.Tracef("Valid range from %s : %s", src.Ip.String(), src.Range.String())
+			}
+		}
+	}
+	if l.Scope != "IP" {
+		src.Scope = l.Scope
+		if _, ok := evt.Meta[l.Scope]; ok {
```

Comment:
I don't think we want to force the scope to be strictly equal to the map key to use, as the user writing the scenario might not have control of the parser that is generating data being processed. For example, `scope` and `scope_source_field` (should the later one even be an expr ?) could be two different fields of the configuration?